src/hooks/useAddressInput.ts

```diff
@@ -0,0 +1,33 @@
+import React from "react";
+import { isValidAddress } from "../utils/address";
+import { useConnection } from "./useConnection";
+
+export function useAddressInput() {
+  const { account } = useConnection();
+
+  const [address, setAddress] = React.useState(account ? account : "");
+  const prevAddress = React.useRef(address);
+
+  const updateAddress = React.useCallback(
+    (newAddress: string) => {
+      prevAddress.current = address;
+      setAddress(newAddress);
+    },
+    [address]
+  );
+
+  React.useEffect(() => {
+    if (!account) return;
+
+    if (!prevAddress.current) {
+      setAddress(account);
+    }
+  }, [account]);
+  const isValid = address ? isValidAddress(address) : true;
```

Comment:
why does it say no address is valid?

Reply:
This is used for input validation, if `address` is an empty string or undefined we don't want to show an error message. However, the Input in the `Hero` is disabled if the address is null and there is null check before it is submitted.